pkg/operator/controllers/internetchecker/internetchecker_controller.go

```diff
@@ -62,64 +71,76 @@ func (r *InternetChecker) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 		return reconcile.Result{}, err
 	}
 
-	var condition status.ConditionType
-	switch r.role {
-	case operator.RoleMaster:
-		condition = arov1alpha1.InternetReachableFromMaster
-	case operator.RoleWorker:
-		condition = arov1alpha1.InternetReachableFromWorker
-	}
+	// limit all checks to take no longer than half of the requeueInterval
+	ctx, cancel := context.WithTimeout(context.Background(), requeueInterval/2)
+	defer cancel()
 
-	urlErrors := map[string]string{}
+	checks := make(map[string]chan error)
 	for _, url := range instance.Spec.InternetChecker.URLs {
-		err = r.check(&http.Client{}, url)
-		if err != nil {
-			urlErrors[url] = err.Error()
-		}
+		checks[url] = make(chan error)
+		go r.checkWithRetry(ctx, &http.Client{}, url, checkBackoff, checks[url])
 	}
 
-	if len(urlErrors) > 0 {
-		sb := &strings.Builder{}
-		for url, err := range urlErrors {
+	sb := &strings.Builder{}
+	checkFailed := false
+
+	for url, ch := range checks {
+		if err = <-ch; err != nil {
+			r.log.Infof("URL %s check failed with error %s", url, err)
 			fmt.Fprintf(sb, "%s: %s\n", url, err)
+			checkFailed = true
 		}
-		err = controllers.SetCondition(r.arocli, &status.Condition{
-			Type:    condition,
+	}
+
+	var condition *status.Condition
+	if checkFailed {
+		condition = &status.Condition{
+			Type:    r.conditionType(),
 			Status:  corev1.ConditionFalse,
 			Message: sb.String(),
 			Reason:  "CheckFailed",
-		}, r.role)
+		}
 	} else {
-		err = controllers.SetCondition(r.arocli, &status.Condition{
-			Type:    condition,
+		condition = &status.Condition{
+			Type:    r.conditionType(),
 			Status:  corev1.ConditionTrue,
 			Message: "Outgoing connection successful.",
 			Reason:  "CheckDone",
-		}, r.role)
+		}
 	}
+
+	err = controllers.SetCondition(r.arocli, condition, r.role)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{RequeueAfter: time.Minute, Requeue: true}, nil
+	return reconcile.Result{RequeueAfter: requeueInterval, Requeue: true}, nil
 }
 
-func (r *InternetChecker) check(client simpleHTTPClient, url string) error {
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-	defer cancel()
-
-	req, err := http.NewRequestWithContext(ctx, http.MethodHead, url, nil)
-	if err != nil {
-		return err
-	}
+// check the URL, retrying failed queries a few times
+func (r *InternetChecker) checkWithRetry(
+	ctx context.Context,
+	client simpleHTTPClient,
+	url string,
+	backoff wait.Backoff,
+	ch chan error,
```

Comment:
use a usual method signature which returns error in order to avoid propagating the channel where it doesn't need to go.  Makes unit testing easier and removes temptation to abuse the channel later in some way.